File: src/routing/KNNTable.cs
using System;
using System.Collections.Generic;

namespace DVAN.Routing
{
    public class NNTable : INNTable
    {
        private (int, float)[] ranges { get; set; }

        public NNTable((int, float)[] ranges)
        {
            this.ranges = ranges;
        }

        public (int, float) getNearest(int destination)
        {
            return this.ranges[destination];
        }
    }

    public class KNNTable : IKNNTable
    {
        private (int, float)[,] ranges { get; set; }

        public KNNTable((int, float)[,] ranges)
        {
            this.ranges = ranges;
        }

        public (int, float) getKNearest(int destination, int k)
        {
            return this.ranges[destination, k];
        }
    }
}

File: src/main/java/org/tud/oas/routing/KNNTable.java
package org.tud.oas.routing;

public class KNNTable implements IKNNTable {
    private int[][] nearest;
    private float[][] ranges;

    public KNNTable(int[][] nearest, float[][] ranges) {
        this.nearest = nearest;
        this.ranges = ranges;
    }

    public int getKNearest(int destination, int k) {
        return this.nearest[destination][k];
    }

    public float getKNearestRange(int destination, int k) {
        return this.ranges[destination][k];
    }
}